packages/fex/fex.patch

```diff
@@ -0,0 +1,210 @@
++++ b/FEXCore/Source/Utils/Allocator/IntrusiveArenaAllocator.h
+@@ -9,19 +9,21 @@
+ 
+ #include <bitset>
+ #include <cstddef>
+-#ifdef TERMUX_BUILD
++
+ #ifdef __has_include
+ #if __has_include(<memory_resource>)
+-#error Termux <experimental/memory_resource> workaround can be removed
+-#endif
+-#endif
++#include <memory_resource>
++namespace fex_pmr = std::pmr;
++#else
+ #include <experimental/memory_resource>
+ #include <experimental/list>
+ namespace fex_pmr = std::experimental::pmr;
++#endif
+ #else
+ #include <memory_resource>
+ namespace fex_pmr = std::pmr;
+ #endif
++
+ #include <sys/user.h>
+ 
+ #include <mutex>
+diff --git a/FEXCore/include/FEXCore/Utils/AllocatorHooks.h b/FEXCore/include/FEXCore/Utils/AllocatorHooks.h
+index 85d2d914d..9997d03cb 100644
+--- a/FEXCore/include/FEXCore/Utils/AllocatorHooks.h
++++ b/FEXCore/include/FEXCore/Utils/AllocatorHooks.h
+@@ -117,16 +117,37 @@ namespace FEXCore::Allocator {
+   inline void *valloc(size_t size)
+   {
+ #ifdef __ANDROID__
++#if __ANDROID_API__ < 28
++    // https://github.com/llvm/llvm-project/blob/main/libcxx/include/__memory/aligned_alloc.h
++    // https://android.googlesource.com/platform/bionic/+/main/libc/platform/bionic/page.h
++    // alignment = 1 segfault in Android, 4096 is the minimum
++    void* __result = nullptr;
++    (void)::posix_memalign(&__result, 4096, size);
++    return __result;
++#else
+     return ::aligned_alloc(4096, size);
++#endif
+ #else
+     return ::valloc(size);
+ #endif
+   }
++#ifdef __ANDROID__
++  inline int posix_memalign(void** r, size_t a, size_t s) { return ::posix_memalign(r, 4096, s); }
++#else
+   inline int posix_memalign(void** r, size_t a, size_t s) { return ::posix_memalign(r, a, s); }
++#endif
+   inline void *realloc(void* ptr, size_t size) { return ::realloc(ptr, size); }
+   inline void free(void* ptr) { return ::free(ptr); }
+   inline size_t malloc_usable_size(void *ptr) { return ::malloc_usable_size(ptr); }
++#if defined(__ANDROID__) && __ANDROID_API__ < 28
++  inline void *aligned_alloc(size_t a, size_t s) {
++    void* __result = nullptr;
++    (void)::posix_memalign(&__result, 4096, s);
++    return __result;
++  }
++#else
+   inline void *aligned_alloc(size_t a, size_t s) { return ::aligned_alloc(a, s); }
++#endif
+   inline void aligned_free(void* ptr) { return ::free(ptr); }
+ #endif
+ 
+diff --git a/Source/Common/Config.cpp b/Source/Common/Config.cpp
+index 5d6cfed33..74918edd1 100644
+--- a/Source/Common/Config.cpp
++++ b/Source/Common/Config.cpp
+@@ -371,6 +371,7 @@ namespace JSON {
+     FEXCore::Config::Load();
+ 
+     auto Args = FEX::ArgLoader::Get();
++	size_t i = 0;
+ 
+     if (LoadProgramConfig) {
+       if (Args.empty()) {
+@@ -378,8 +379,19 @@ namespace JSON {
+         return {};
+       }
+ 
+-      Args[0] = RecoverGuestProgramFilename(std::move(Args[0]), ExecFDInterp, ProgramFDFromEnv);
+-      fextl::string& Program = Args[0];
++      // really bad workaround for Termux Proot
```

Comment:
Unfortunately this "hack" still doesn't work with proot. The FEXServer seems to be getting libc++abi error which I can't figure out why. This is still the case in the old version. May even exist since the first version packaged for Termux.

I also doubt there is people actually using this as the various guide I see just use FEX inside proot distro. Only FEXIntepreter with x86_64 musl static busybox is working. If no one want to improve this, I suggest move to disabled.